### PR TITLE
Grid boxes

### PR DESCRIPTION
**Pull request by Alexandru-Bulai**
# PR 23-01-24
## Overview
In this PR i added all the missing images and boxes on the grid, each of them with different rating and gender, also made the design responsive across different devices
## Key Changes :
 1) Grid: 
    - Added images and boxes nedded on the grid
    - For each image i changed gender and rating 
    - Made the grid responsive across devices 
    - Made the background image contain the elements from the grid 

Jan-23-2024 19-59-26

Jan-23-2024 20-08-40

### CODE REVIEW

File: public/gallery.html [hunk 1]
```diff
@@ -106,12 +132,180 @@
...
+
 <!-- Menu end -->
-<div>
-	<img  src="Assets/images/gallery/percare_gallery_hero.png" alt="Animals posing in a art gallery">
+
+<!-- Grid section -->
+<div id="gridBackground">
+
+<!-- Grid container start -->
+	<div class="w-full h-full grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4" id="gridContainer">
```
Comment (vladiiancu) [inline public/gallery.html]: The ID is way too generic and specific to a css property as well

File: public/gallery.html [hunk 2]
```diff
@@ -106,12 +132,180 @@
...
-<!-- End filters container -->
 </div>
+<!-- End filters container -->
+
 <!-- Menu end -->
-<div>
-	<img  src="Assets/images/gallery/percare_gallery_hero.png" alt="Animals posing in a art gallery">
+
+<!-- Grid section -->
+<div id="gridBackground">
```
Comment (vladiiancu) [inline public/gallery.html]: Name is tied to CSS property. What happens if we decide that grid is not a good option and we need to change to flex?

File: public/gallery.html [hunk 3]
```diff
@@ -7,6 +7,55 @@
 	</head>
 <body class="bg-blue-50">
 <script src="https://cdn.tailwindcss.com"></script>
+<style>
```
Comment (vladiiancu) [inline public/gallery.html]: Why are we having so much style inlined in a page?

File: public/gallery.html [hunk 4]
```diff
@@ -106,12 +132,180 @@
...
+
 <!-- Menu end -->
-<div>
-	<img  src="Assets/images/gallery/percare_gallery_hero.png" alt="Animals posing in a art gallery">
+
+<!-- Grid section -->
+<div id="backgroundImg">
+
+<!-- Grid container start -->
+	<div class="w-full h-full grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4" id="imgContainer">
```
Comment (vladiiancu) [inline public/gallery.html]: So the entire website will only ever have a single imgContainer? 

This is what results from the combination of name and id. 

imgContainer is generic. There could be many, multiple image containers in a page and in an entire website. 
So this name would be more appropriate as a class. 

So you could have multiple elements with the class imgContainer and then each element will have its own id which differences through its purpose. In this case, petGallery might be an idea. 

Now that I think about it, this element contains a bunch of images. imgContainer usually means "image container" so this leads me to a single image. Not multiple. 

I see e changes required here

File: public/gallery.html [hunk 5]
```diff
@@ -106,12 +132,180 @@
...
+
+		<div class="imageHoverContainer relative">
+			<img src="Assets/images/gallery/gallery_grid/cat-1.png" alt="dog portrait">
+			<div class="statsBox flex flex-col justify-center items-center  absolute bottom-0 inset-x-0 bg-purple-400 h-24 bg-opacity-90 opacity-0">
+				<div class="text-15 font-bold text-blue-800">Gender: male</div>
+				<div class="text-15 font-bold text-blue-800">Rating: 3☆</div>
+			</div>
+		</div>
+
+		<div class="imageHoverContainer relative">
```
Comment (vladiiancu) [inline public/gallery.html]: Again this name is too tight with your css intentions. 

So for a mobile device, what does the class image hover do? There is no hover on mobile. 

Also in one place you are using "image" and in another it's "img"

File: public/gallery.html [hunk 6]
```diff
@@ -4,9 +4,11 @@
 		<meta charset="UTF-8">
 		<meta name="viewport" content="width=device-width, initial-scale=1.0">
 		<title>Pet Gallery</title>
+		<link rel="stylesheet" href="CSS/gallery.css">
```
Comment (omniconvert) [inline public/gallery.html]: CSS is cascading. So if you first put this and then tailwind you risk having things overwritten by it

**Review by omniconvert — COMMENTED** [review]
Nic description. Useful images

**Review by vladiiancu — DISMISSED** [review]

**Review by vladiiancu — DISMISSED** [review]

**Review by vladiiancu — DISMISSED** [review]

**Review by omniconvert — DISMISSED** [review]

**Review by omniconvert — DISMISSED** [review]
k

**Review by vladiiancu — COMMENTED** [review]
K

**Review by vladiiancu — DISMISSED** [review]

**Pull request closed without merging**